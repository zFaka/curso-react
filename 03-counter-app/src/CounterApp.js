import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( {value} ) => {

    const [counter, setCounter] = useState(value);

    const handlerClickUp = () => {
        setCounter(counter+1);
    }

    const handlerClickDown = (  ) => {
        setCounter(counter-1);
    }

    const handlerClickRst = () => {
        setCounter(value);
    }


    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <h3>
                <button onClick={handlerClickDown}>-</button>
                <button onClick={handlerClickRst}>R</button>
                <button onClick={handlerClickUp}>+</button>
            </h3>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;


