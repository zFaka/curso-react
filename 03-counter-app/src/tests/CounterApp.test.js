import React from 'react';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('CounterApp test', () => {

    let renderOutput;

    beforeEach(() => {

        renderOutput = shallow(<CounterApp value={0} />);

    })

    test('Should be equal to snapshot', () => {

        // shallow should be render the Component with the assigned values in test
        const renderOutput = shallow(<CounterApp value={0} />);

        expect(renderOutput).toMatchSnapshot();

    })

    test('Should show default value of 100', () => {

        // shallow should be render the Component with the assigned values in test
        const renderOutputCien = shallow(<CounterApp value={100}/>);

        // trim remove the spaces. without strim the output is <Space>100<Space>
        const outputValue = renderOutputCien.find('h2').text().trim();

        expect(outputValue).toBe('100');

    })

    test('Should increase the value of +1 button', () => {

        //const btn1 = renderOutputButton.find('button').at(2);
        // btn1 value is <button>+</button>

        const renderOutputButton = shallow(<CounterApp value={0}/>);
        renderOutputButton.find('button').at(2).simulate('click');
        const counterText = renderOutputButton.find('h2').text();

        expect(counterText).toBe('1');

    })

    test('Should decrease the value og -1 button', () => {

        const renderOutputButton = shallow(<CounterApp value={0}/>);
        renderOutputButton.find('button').at(0).simulate('click');
        const counterText = renderOutputButton.find('h2').text();

        expect(counterText).toBe('-1');

    })

    test('Should set reset value', () => {

        const renderOutput= shallow(<CounterApp value={100}/>);

        renderOutput.find('button').at(2).simulate('click');
        renderOutput.find('button').at(2).simulate('click');
        renderOutput.find('button').at(1).simulate('click');
        const counterText = renderOutput.find('h2').text();

        expect(counterText).toBe("100");

    })

})
