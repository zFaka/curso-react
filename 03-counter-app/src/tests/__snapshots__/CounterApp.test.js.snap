// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp test Should be equal to snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <h3>
    <button
      onClick={[Function]}
    >
      -
    </button>
    <button
      onClick={[Function]}
    >
      R
    </button>
    <button
      onClick={[Function]}
    >
      +
    </button>
  </h3>
</Fragment>
`;
